### DISCUSSION TITLE
gzip buildManifest and ssgManifest

### DISCUSSION THREAD

**Discussion by jaybytez** [vercel/next.js#18685] [Help]
Looks like most of the files out of Next.js are gzipped when running npm run build/start.  But buildManifest and ssgManifest are not.  Is it possible to gzip these two files, maybe through the headers feature?

**Comment by jaybytez** [comment]
I tried to apply Content-Encoding to gzip for these assets, but even though I apply the source paths via the headers, it doesn't work.  https://nextjs.org/docs/api-reference/next.config.js/headers.  Is there documentation missing as to what can and cannot be applied through these headers.

**Reply by kumarldh** [reply]
@jaybytez for me only the ssgManifest.js file had this issue. I am using a custom server set up and for all the files I was able to add custom headers. However, `ssgManifest.js` is just 77 bytes and adding cache headers for it w/out adding specific handling for this file has been very difficult.

**Reply by kumarldh** [reply]
@jaybytez I disabled compression from next js. Here is how to do it https://nextjs.org/docs/api-reference/next.config.js/compression. I also disabled etag generation. Then I added following to my nginx conf
```
        # enable gzip compression
        gzip on;
        gzip_min_length  10;
        gzip_buffers  4 32k;
        gzip_types    text/plain text/html text/xml text/css application/javascript application/json;
        gzip_vary on;
        # end gzip configuration
```

Notice the small value for `gzip_min_length`.  Nginx wont compress files which have size less than `gzip_min_length`. I found the default was 1K for the docker image I had. In case you are using Apache, it has similar behavior. You may have to tune it accordingly.

**Reply by AshokChava** [reply]
Looking at the files I see that any file that is not more than 2KB is not being compressed. So wondering if there is a config in next.config.js to make sure all the files are gzip. The biggest issue if the files are not gziped our CDN is not caching them, we want to make sure that all the files are cached by CDN.

**Reply by kumarldh** [reply]
@AshokChava I faced exactly same issue. Please disable GZIP in Next and then enable GZIP at web server level. I use NGINX and you can refer to my solution. HTH!

**Reply by MohtashimAli85** [reply]
Hi, I am stuck in this kind of error.
here is my nginx file
![image](https://user-images.githubusercontent.com/68479602/165512001-0fbf6b19-cbd4-4c62-af0a-99627faff793.png)
and here is next config
![image](https://user-images.githubusercontent.com/68479602/165512113-3c6f9dae-9232-4b0c-a80f-498bd5f575fe.png)
facing this error
![image](https://user-images.githubusercontent.com/68479602/165512157-1a418822-bce1-4ff7-a65f-2167abb12e2c.png)
